src/types.rs

```diff
@@ -236,12 +213,24 @@ impl Tile {
 }
 
 #[derive(Debug, Clone)]
-pub struct GameWindow {
-    pub world: World,
+pub struct SokoModel {
+    pub current_state: World,
     pub zoom: Zoom,
+    pub history: Vec<World>,
     pub debug: Vec<String>,
 }
 
+impl SokoModel {
+    pub fn from(world: World) -> Self {
```

Comment:
This from statement is kinda about the react state conversation topic we had in chat, right? This looks good.